mantidproject/mantid
# ConvertUnits now works with Point Workspaces

## Description of work
- `ConvertUnits` now accepts workspaces with Point data.
- `ConvertUnits` has a new property that toggles that functionality on and off
- 2 new Unit Tests have been added to test the new functionality with Point data
- A new Performance Test has been added to test the new functionality with Point data
- The InputWorkspace is converted to Bins, and, at the end, back to Points
## To test:

I used the workspace [here](https://drive.google.com/file/d/0B2Mj8e56Z8xTZjNSNFNScHhBT2c/view?usp=sharing) <- can also be found from the TrainingCourseData, with the helper Python Script found [here](http://pastebin.com/mSwMzx2L)

If you want to try without the fix:
- Load the `164199` workspace into Mantid
- Run the helper Python Script
- It should fail on `ConvertUnits(points_ws, Target='Wavelength')` as `ConvertUnits` doesn't accept Point Data

Testing after applying the fix:
- Load the `164199` workspace into Mantid
- Run the helper Python Script
- Everything should be completed successfully
- You can now test the `ConvertUnits` manually from the `Algorithms` window
- The `ConvertUnits` dialogue has a new option: `ConvertFromPointData`, set by default to `True`
- Check the tooltip text of the checkbox for spelling mistakes and clarity
- Try converting a Points workspace with the option unchecked
- This should fail and an exception should be shown in the log, check the message for spelling mistakes and clarity
- Try again with the option checked, it should now run successfully

Additional:
- Run the `ConvertUnits` unit test. There are 2 new tests. All tests must pass
- Run the `ConvertUnits` performance test. There is 1 new test. All tests must pass. (they take a long time, they might need to be reworked, as they currently load files in)
- Check the documentation for spelling mistakes/clarity. The documentation has been changed both for the release notes (link) and for the algorithm itself.
- Build the documentation and check for bad formatting, any spelling mistakes or anything unclear.
- Check the code quality/readability



Fixes #13280.



Added in release notes [here](https://github.com/mantidproject/mantid/pull/17266/files#diff-fef8ac613fa6d94a43015ee70c230735R69)

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
